pkgs/games/lunar-client/default.nix

```diff
@@ -1,45 +1,37 @@
-{ appimageTools, lib, fetchurl, makeDesktopItem }:
+{ appimageTools
+, fetchurl
+, lib
+}:
 
 let
-  name = "lunar-client";
-  version = "2.15.1";
-
-  desktopItem = makeDesktopItem {
-    name = "lunar-client";
-    exec = "lunar-client";
-    icon = "lunarclient";
-    comment = "Minecraft 1.7, 1.8, 1.12, 1.15, 1.16, 1.17, and 1.18 Client";
-    desktopName = "Lunar Client";
-    genericName = "Minecraft Client";
-    categories = [ "Game" ];
-  };
-
-  appimageContents = appimageTools.extract {
-    inherit name src;
-  };
+  pname = "lunar-client";
+  version = "3.0.6";
+  name = "${pname}-${version}";
 
   src = fetchurl {
     url = "https://launcherupdates.lunarclientcdn.com/Lunar%20Client-${version}.AppImage";
-    name = "lunar-client.AppImage";
-    hash = "sha256-8F6inLctNLCrTvO/f4IWHclpm/6vqW44NKbct0Epp4s=";
+    hash = "sha256-ujDiGfs2py/Sanh1sgXGLfP4ergm3O4p90B4M3WgU54=";
   };
+
+  appimageContents = appimageTools.extract { inherit name src; };
 in
-appimageTools.wrapType1 rec {
-  inherit name src;
+appimageTools.wrapType2 rec {
+  inherit pname version src;
 
   extraInstallCommands = ''
-    mkdir -p $out/share/applications
-    cp ${desktopItem}/share/applications/* $out/share/applications
-    cp -r ${appimageContents}/usr/share/icons/ $out/share/
+    mv $out/bin/{${name},${pname}}
+    install -Dm444 ${appimageContents}/launcher.desktop $out/share/applications/${pname}.desktop
+    install -Dm444 ${appimageContents}/launcher.png $out/share/pixmaps/${pname}.png
+    substituteInPlace $out/share/applications/${pname}.desktop \
+      --replace 'Exec=AppRun --no-sandbox %U' 'Exec=${pname}' \
+      --replace 'Icon=launcher' 'Icon=${pname}'
```

Comment:
We shouldn't be using pname here so often

Reply:
Should I just write lunar-client here explicitly?

Reply:
yeah

Reply:
done!